booster/test/rpc-integration/test-log-simplify-json/simplify-log.txt.golden

```diff
@@ -2,88 +2,6 @@
 {"tag":"simplification","result":{"tag":"success","rule-id":"f310443b160236f56d095576a7500c11d1e98ee7af349b0bbf7a34b5e444d300"},"origin":"booster"}
 {"tag":"simplification","result":{"tag":"failure","reason":"RuleNotPreservingDefinedness","rule-id":"3b51b2998e8ebe79b88550ea9d27ca95003744019b16b93d17a55f6f9bcaa12a"},"origin":"booster"}
 {"tag":"simplification","result":{"tag":"failure","reason":"RuleNotPreservingDefinedness","rule-id":"1170b8c2f6d770eecb12b2ffb28275ed94ae1451d0ae3f2b42e570cd198099fa"},"origin":"booster"}
-{"tag":"simplification","result":{"tag":"failure","reason":"Failed match","rule-id":"c2682318552299808618ea1106c22cf9df82485521eaf5e58844a16e0d4550e4"},"origin":"booster"}
```

Comment:
Am I right to understand that we are evaluating certain things with the hooks, whereas before we attempted to apply rules from `domains.md`?

Reply:
Correct, the hook is tried first (if one exists), and [if the hook returns a result, no equations are tried](https://github.com/runtimeverification/haskell-backend/blob/c42b8f0f9b7a005429c3d59e8c3e8eeeac1b2155/booster/library/Booster/Pattern/ApplyEquations.hs#L686-L687).